crates/oxc_linter/src/rules/unicorn/prefer_set_has.rs

```diff
@@ -0,0 +1,900 @@
+use itertools::Itertools;
+use oxc_ast::{
+    ast::{Expression, MemberExpression, VariableDeclarationKind},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::ScopeId;
+use oxc_span::Span;
+use phf::phf_set;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+const ARRAY_METHODS_RETURNS_ARRAY: phf::Set<&'static str> = phf_set! {
+   "concat",
+    "copyWithin",
+    "fill",
+    "filter",
+    "flat",
+    "flatMap",
+    "map",
+    "reverse",
+    "slice",
+    "sort",
+    "splice",
+    "toReversed",
+    "toSorted",
+    "toSpliced",
+    "with",
+};
+
+fn prefer_set_has_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Switch to `Set`")
+        .with_help("should be a `Set`, and use .has()` to check existence or non-existence.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferSetHas;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prefer `Set#has()` over `Array#includes()` when checking for existence or non-existence.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Set#has() is faster than Array#includes().
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const array = [1, 2, 3];
+    /// const hasValue = value => array.includes(value);
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// const set = new Set([1, 2, 3]);
+    /// const hasValue = value => set.has(value);
+    /// ```
+    /// ```js
+    /// const array = [1, 2, 3];
+    /// const hasOne = array.includes(1);
+    /// ```
+    PreferSetHas,
+    perf,
+    dangerous_fix
+);
+
+fn is_includes_call(member_expr: &MemberExpression) -> bool {
+    if member_expr.is_computed() || member_expr.optional() {
+        return false;
+    }
+
+    let Some(callee_name) = member_expr.static_property_name() else {
+        return false;
+    };
+
+    callee_name == "includes"
+}
+
+fn is_array_of_or_from(name: &str, callee: &MemberExpression) -> bool {
+    if name != "from" && name != "of" {
+        return false;
+    }
+
+    return callee.object().get_identifier_reference().map_or(false, |ident| ident.name == "Array");
+}
```

Comment:
would `is_specific_member_access` work here?

```suggestion
fn is_array_of_or_from(callee: &MemberExpression) -> bool {
    callee.is_specific_member_access("Array", "of") || callee.is_specific_member_access("Array", "from")
}
```

Reply:
done